### PR TITLE
feat: optional tx params

### PR DESCRIPTION
**Pull request by QuinnLee**
Pass TX params to transfer
```
gasLimit, gasPrice, bytePrice, maturity
```

### CODE REVIEW

File: packages/wallet/src/transfer.test.ts [hunk 1]
```diff
@@ -19,6 +19,34 @@ describe('Wallet', () => {
...
+      await sender.transfer(receiver.address, 1, NativeAssetId, {
+        gasLimit: 1,
+        gasPrice: 1,
+        bytePrice: 1,
+        maturity: 1,
+      });
+    } catch (e) {
+      const error = e as Error;
+      expect(error.message).toContain('InsufficientGas');
+    }
```
Comment (QuinnLee) [inline packages/wallet/src/transfer.test.ts:39]: tried to do `expect().toThrow()` but didnt work.  Is there another way?

Comment (luizstacio) [reply packages/wallet/src/transfer.test.ts:39]: expect(async () => {
      await sender.transfer(receiver.address, 1, NativeAssetId, {
        gasLimit: 1,
        gasPrice: 1,
        bytePrice: 1,
        maturity: 1,
      });
}).toThrow....

Comment (QuinnLee) [reply packages/wallet/src/transfer.test.ts]: Figured it out! Didn't realize you have to `await` the expect

```
   await expect(async () => {
      await sender.transfer(receiver.address, 1, NativeAssetId, {
        gasLimit: 1,
        gasPrice: 1,
        bytePrice: 1,
        maturity: 1,
      });
    }).rejects.toThrowError('InsufficientGast');
```

**Review by AlicanC — APPROVED** [review]

File: packages/wallet/src/wallet.ts [hunk 2]
```diff
@@ -191,9 +191,12 @@ export default class Wallet extends AbstractWallet {
     /** Amount of coins */
     amount: BigNumberish,
     /** Asset ID of coins */
-    assetId: BytesLike = NativeAssetId
+    assetId: BytesLike = NativeAssetId,
+    /** Tx Params */
+    txParams: Pick<TransactionRequestLike, 'gasLimit' | 'gasPrice' | 'bytePrice' | 'maturity'> = {}
   ): Promise<TransactionResponse> {
-    const request = new ScriptTransactionRequest({ gasLimit: 1000000 });
+    const params = { gasLimit: 10000, ...txParams };
```
Comment (camsjams) [inline packages/wallet/src/wallet.ts:198]: Was the change in magnitude of the gasLimit was intentional? 
`1000000` => `10000`

Comment (camsjams) [reply packages/wallet/src/wallet.ts:198]: @QuinnLee ^

**Pull request merged**